### DISCUSSION TITLE
Getting errors when running the virtual_webcam.py script

### DISCUSSION THREAD

**Discussion by naowalrahman** [allo-/virtual_webcam_background#60] [General]
While trying to install this, everything was working until I ran virtual_webcam.py to start the program. For some reason, I always get this error: 
![image](https://user-images.githubusercontent.com/67881983/106323001-b17d2900-6244-11eb-933a-03cb183e4ede.png)
Any ideas as to why, and the fixes for it?

**Comment by kevingatera** [comment]
Same here, maybe there's something we're doing wrong?

**Comment by allo-** [comment]
Looks like you do not have the rights to access the v4l2loopback device or specified the wrong device in the configuration.

Please check:

- What parameters did you use when loading v4l2loopback? video_nr=X should correspond to the device number
- Is `virtual_video_device` the device created when loading the v4l2loopback kernel module?
- Is your user account allowed to write to the device?

**Comment by kevingatera** [comment]
Yeah you're right. Turns out the virtual_video_device in config.yaml was wrong. I tried each of the video devices in /dev until I found the right one. I am trying to add a background to a droidcam webcam feed, and it works! Albeit, slowly, very slowly.

**Comment by allo-** [comment]
I guess it's time for a FAQ and Troubleshooting section.

**Comment by naowalrahman** [comment]
Ok, so I am the admin account, so it's definitely allowed to write to the device. How do I set parameters while loading v4l2loopback, and how do I check if `virtual_video_device` is created (sorry I'm kind of a newbie to this stuff)?

**Comment by allo-** [comment]
As root:

```
modprobe v4l2loopback exclusive_caps=1 video_nr=10
ls -l /dev/video*
```

You should now see a `video10` device and your webcam, if it is plugged in. Have a look at the group name (e.g. "video")
As a user check with `groups` if you're in the group. Otherwise add your user to the group and re-login.

Now you should be able to use video10 as virtual device and the webcam device as real device using your normal user account.
